scoring.js

```diff
@@ -0,0 +1,81 @@
+class Scoring {
+  constructor(scorecard) {
+    this.scorecard = scorecard
+    this.bonusScore = []
+    this.regularScore = []
+    this.totalScorePerFrame = []
+    this.totalScore = null
+    this.n = 0
+  }
+  regularScorePerFrame() { 
+    // this is just the number of pins per frame, though frame 10 is an exception
+    this.regularScore = this.scorecard.map( element => element.reduce(function(a,b){ return a + b }, 0) )
+    if ( this.scorecard[9][0] === 10) { this.regularScore[9] = 10 }
+    else { this.regularScore[9] = this.scorecard[9][0] + this.scorecard[9][1] }
+  }
+  bonusScore8Frames() {
+    var frameBonus = 0
+    while ( this.n < 8 ) {
+      // 3 or 2 strikes in a row
+      if ( this.scorecard[this.n][0] == 10 && this.scorecard[this.n + 1 ][0] == 10 ) {
```

Comment:
maybe you could break these ifs into separate methods to make each method cleaner with fewer responsibilities.